docs/source/cuda/cuda_array_interface.rst

```diff
@@ -76,24 +76,376 @@ The following are optional entries:
+    provided stream, then it may return control to the User code immediately
+    after enqueueing its work, as the work will all be serialized on the
+    exported array's stream. This is sufficient to ensure correctness even if
+    the User code were to induce the Producer to subsequently start enqueueing
+    more work on the same stream.
+
+* If the User has set the Consumer to ignore CAI synchronization semantics, the
+  Consumer may assume it can operate on the data immediately in any stream with
+  no further synchronization, even if the ``stream`` member has an integer
+  value.
+
+
+When exporting an array through the CAI, Producers must ensure that:
+
+* If there is work on the data enqueued in one or more streams, then
+  synchronization on the provided ``stream`` is sufficient to ensure
+  synchronization with all pending work.
+
+  - If the Producer has no enqueued work, or work only enqueued on the stream
+    identified by ``stream``, then this condition is met.
+  - If the Producer has enqueued work on the data on multiple streams, then it
+    must enqueue events on those streams that follow the enqueued work, and
+    then wait on those events in the provided ``stream``. For example:
+
+    1. Work is enqueued by the Producer on streams ``7``, ``9``, and ``15``.
+    2. Events are then enqueued on each of streams ``7``, ``9``, and ``15``.
+    3. Producer then tells stream ``3`` to wait on the events from Step 2, and
+       the ``stream`` entry is set to ``3``.
+
+* If there is no work enqueued on the data, then the ``stream`` entry may be
+  either ``None``, or not provided.
+
+Optionally, to facilitate the User to relax the conformance to synchronization
+semantics:
+
+* Producers may provide a configuration option to always set ``stream`` to
+  ``None``.
+* Consumers may provide a configuration option to ignore the value of ``stream``
+  and act as if it were ``None`` or not provided.  This elides synchronization
+  on the Producer-provided streams, and allows enqueuing work on streams other
+  than that provided by the Producer.
+
+These options should not be set by default in either a Producer or a Consumer.
+The exact mechanism by which these options are set, and related options that
+Producers or Consumers might provide to allow the user further control over
+synchronization behavior are not prescribed by the CAI specification.
+
+
+Synchronization in Numba
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Numba is neither strictly a Producer nor a Consumer - it may be used to
+implement either by a User. In order to facilitate the correct implementation of
+synchronization semantics, Numba exhibits the following behaviors related to
+synchronization of the interface:
+
+- When Numba acts as a Consumer (for example when an array-like object is passed
+  to a kernel launch): If ``stream`` is an integer, then Numba will immediately
+  synchronize on the provided ``stream``. A Numba Device Array created from an
```

Comment:
I tried to avoid directly indicating a `DeviceArray`, because it might also be a mapped or managed array. Perhaps "Numba CUDA Array" will be clearer without being overly prescriptive as to the type?

Reply:
think that's a good idea, as this isn't just about device arrays.

Reply:
On second thoughts, Numba always creates a `DeviceNDArray` when consuming, so it does make sense to link to it.

Reply:
Added a link to the `DeviceNDArray` - some edits to the later text were also added (sometimes changing "Device Array" to "CUDA Array" or just "array") to ensure the section as a whole is clear and consistent.